.github/workflows/checks.yml

```diff
@@ -8,6 +8,10 @@ on:
   schedule:
     - cron: '0 23 * * SUN-THU'
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
```

Comment:
It limits one concurrency for each PR and master branch.

Reply:
If we want to limit one concurrency for each PR, but multiple concurrencies to master. We can use the following:

```
${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
```

Reply:
I prefer the last one: one concurrency for each PR but multiple ones to master. This is because it is easier to identify the cause of CI failures when PRs are merged continuously in a short period of time.

Reply:
> It limits one concurrency for each PR and master branch.

Let me check my understanding. Since the `group` includes the [`github.workflow` context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context), this `concurrency` rule limits one concurrency *for each workflow*, right? That means we run at most 13 concurrent jobs per PR because we have 13 different workflows.

Reply:
> Since the group includes the github.workflow context, this concurrency rule limits one concurrency for each workflow, right? That means we run at most 13 concurrent jobs per PR because we have 13 different workflows.

Yes. In my understanding, that will be what we want to do.